applications/postbuildstepper/src/lib.rs

```diff
@@ -502,30 +507,39 @@ pub mod business {
                     let source_branch_channels = HashSet::<&str>::from_iter(
                         build_info.get(SOURCE_BRANCH_CHANNELS_ENV_KEY)?.split(","),
                     );
-                    debug!("will create channels for these source branches: {source_branch_channels:#?}");
-                    let mut channel_names = vec![number];
+
+                    debug!("configured PR source branches for channel creation: {source_branch_channels:#?}");
+
                     if source_branch_channels.contains(source_branch.as_str()) {
-                        channel_names.push(source_branch);
-                    };
-                    channel_names
+                        // also include the PR number as a channel name
+                        Some(vec![number, source_branch])
+                    } else {
+                        None
+                    }
                 }
                 BuildInfoEvent::Push {
                     destination_branch, ..
-                } => vec![destination_branch],
+                } => Some(vec![destination_branch]),
             };
 
-            let revision = build_info.try_revision()?;
+            debug!("channel names for this build: {maybe_channel_names:#?}");
 
-            create_channel_artifacts(
-                revision,
-                channel_names.as_slice(),
-                pbs_channels_directory,
-                tarball_path,
-                pbs_jobsets_directory,
-            )?;
-        }
+            if let Some(channel_names) = &maybe_channel_names {
+                let revision = build_info.try_revision()?;
 
-        Ok(())
+                create_channel_artifacts(
+                    revision,
+                    channel_names.as_slice(),
+                    pbs_channels_directory,
+                    tarball_path,
+                    pbs_jobsets_directory,
+                )?;
+            }
+
+            Ok(maybe_channel_names)
+        } else {
+            Ok(None)
```

Comment:
Should we add some kind of log or notification hook for whenever this occurs?

Reply:
there's a `debug!` one that prints this further up.
further down the `fn create_channel_artifacts` has `info!` level messages for artifacts that are actually created.

do you think that's sufficient?

Reply:
I saw the debug a bit earlier, which could be fine for now... but depending on how long we need to support the legacy channels, it might be nice to have some other mechanism that notifies us more directly or loudly, such as via a monitoring tool dashboard like Grafana or Sentry or even a mm plugin, etc.

Reply:
> it might be nice to have some other mechanism that notifies us more directly or loudly than developer logs, such as push messages, or emails via a monitoring tool like Grafana or Sentry or even a mm plugin, etc.

agreed, however out of scope for this bugfix PR.